src/core_write.cpp

```diff
@@ -88,7 +88,12 @@ string ScriptToAsmStr(const CScript& script, const bool fAttemptSighashDecode)
         }
         if (0 <= opcode && opcode <= OP_PUSHDATA4) {
             if (vch.size() <= static_cast<vector<unsigned char>::size_type>(4)) {
-                str += strprintf("%d", CScriptNum(vch, false).getint());
+                if ( (lastopcode == OP_NAME_NEW || lastopcode == OP_NAME_UPDATE || lastopcode == OP_NAME_FIRSTUPDATE )
+                    && vch.size() > static_cast<vector<unsigned char>::size_type>(0) ) {
```

Comment:
Although.. I suppose with a 0-length *field* in the script, it's actually an empty item. How then would a pure 0-length item be best represented in the asm? Like.. []  maybe? I suppose the fact that the asm is not really invertible (I don't know about any script assemblers for namecoin at any rate,) kind of makes this notion moot.